mani_skill/envs/tasks/control/hopper.py

```diff
@@ -0,0 +1,266 @@
+"""Adapted from https://github.com/google-deepmind/dm_control/blob/main/dm_control/suite/hopper.py"""
+
+import os
+from typing import Any, Dict, Union
+
+import numpy as np
+import torch
+from transforms3d.euler import euler2quat
+
+from mani_skill.agents.base_agent import BaseAgent
+from mani_skill.agents.controllers import *
+from mani_skill.envs.sapien_env import BaseEnv
+from mani_skill.envs.utils import randomization, rewards
+from mani_skill.sensors.camera import CameraConfig
+from mani_skill.utils import common, sapien_utils
+
+# fix new imports later
+from mani_skill.utils.building.ground import build_ground
+from mani_skill.utils.registration import register_env
+from mani_skill.utils.structs.pose import Pose
+from mani_skill.utils.structs.types import (
+    Array,
+    GPUMemoryConfig,
+    SceneConfig,
+    SimConfig,
+)
+
+MJCF_FILE = f"{os.path.join(os.path.dirname(__file__), 'assets/hopper.xml')}"
+
+# params from dm_control
+# Minimal height of torso over foot above which stand reward is 1.
+_STAND_HEIGHT = 0.6
+
+# Hopping speed above which hop reward is 1.
+_HOP_SPEED = 2
+
+
+class HopperRobot(BaseAgent):
+    uid = "hopper"
+    mjcf_path = MJCF_FILE
+
+    @property
+    def _sensor_configs(self):
+        return [
+            # replicated from xml file
+            CameraConfig(
+                uid="cam0",
+                pose=Pose.create_from_pq([0, -2.8, 0], euler2quat(0, 0, np.pi / 2)),
+                width=128,
+                height=128,
+                fov=70 * (np.pi / 180),
+                near=0.01,
+                far=100,
+                entity_uid="torso_dummy_1",
```

Comment:
use `mount=self.robot.links_map["torso_dummy_1"]` instead. This is explicit and better as it references the actual link to mount on. Remove entity_uid (we should be deprecating the use of that in the future).

Reply:
fixed :+1: